### DISCUSSION TITLE
How to limit webrtc UDP connections to a specific port range?

### DISCUSSION THREAD

**Discussion by D1plo1d** [webrtc-rs/webrtc#352] [Q&A] [answered]
I'm trying to limit webrtc UDP connections to a port range so that they can be NAT port forwarded by users of an IOT device. So far I haven't been able to figure this out, does anyone know how to select a port range for webrtc-rs?

I see in `/ice/src/udp_networks.rs:7` that EphemeralUDP supports providing a port range and that looked promising at first but I don't see how I could set that from outside the crate boundary.

**Comment by k0nserv** [comment, marked as answer]
This can be configured via `SettingEngine::set_udp_network` as follows

```rust
let mut ephemeral_udp = EphemeralUDP::default();
ephemeral_udp.set_ports(4000, 4050)?;

let mut setting_engine = SettingEngine::default();
setting_engine.set_udp_network(UDPNetwork::Ephemeral(ephemeral_udp));

let api = APIBuilder::new()
    .with_setting_engine(setting_engine)
    .build();
```

EDIT: Updated to use public API so it actually compiles.

**Reply by D1plo1d** [reply]
Thanks @k0nserv, that's what I was missing!

For anyone who stumbles onto this issue later, I modified the example a bit to use `set_ports` because `port_min` and `port_max` aren't public:

```rust
    let mut ephemeral_udp = EphemeralUDP::default();
    ephemeral_udp.set_ports(4000, 4050)?;

    let mut setting_engine = SettingEngine::default();
    setting_engine.set_udp_network(UDPNetwork::Ephemeral(ephemeral_udp));

    let api = APIBuilder::new()
        .with_setting_engine(setting_engine)
        .build();

```

**Reply by k0nserv** [reply]
Thanks. I updated the answer per your comment, as you can tell I wasn't actually compiling the code :)

**Reply by bjungs** [reply]
Currently, the following snippet:

```rust
let mut ephemeral_udp = EphemeralUDP::default();
ephemeral_udp.set_ports(4000, 4050)?;
```

... is currently equivalent to

```rust
let ephemeral_udp = EphemeralUDP::new(4000, 4050)?;
```
